src/Symfony/Component/Validator/Constraints/Encoding.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Validator\Constraints;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\Exception\ConstraintDefinitionException;
+
+/**
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ */
+#[\Attribute(\Attribute::TARGET_PROPERTY | \Attribute::TARGET_METHOD | \Attribute::IS_REPEATABLE)]
+class Encoding extends Constraint
+{
+    public const BAD_ENCODING_ERROR = '94c5e58b-f892-4e25-8fd6-9d89c80bfe81';
+
+    protected const ERROR_NAMES = [
+        self::BAD_ENCODING_ERROR => 'BAD_ENCODING_ERROR',
+    ];
+
+    public array|string $encodings = [];
+    public string $message = 'The detected encoding "{{ detected }}" does not match one of the following encoding: {{ encodings }}.';
+
+    public function __construct(array $options = null, string $message = null, array|string $encodings = null, array $groups = null, mixed $payload = null)
```

Comment:
Not sure if we discussed that before, but do we still need to have the. `$options` parameter when we no longer support annotations?

Reply:
I'm not aware of this, but if it is a thing, I'd be glad to remove this artifact from the past :smile:

Reply:
Alright I had a look at CssColor and it seems you're right. I updated accordingly :slightly_smiling_face:

Reply:
maybe we should wait nonetheless (I found #51393 in the meantime)

Reply:
Even if some constraints are already built without `$options`?